### ISSUE TITLE
Build log gets out to poudriere's terminal output for at least one port: lang/quilc

### ISSUE THREAD

**Issue by yurivict** [freebsd/poudriere#1008]
## Describe the bug
When ```lang/quilc``` is built its build log spills out and gets to the terminal.
I didn't observe this problem with any other port.

## Screenshots
![image](https://user-images.githubusercontent.com/271906/182485773-831967fd-2555-42f9-81c3-bea5eb80403d.png)

## Environment
 - Host OS: 13.1-STABLE amd64
 - Jail OS: 13.1-STABLE amd64
 - Poudriere Version: 3.3.7_1

**Comment by bdrewery** [comment]
I don't know what happened there but I couldn't reproduce it on 3.3.7.
```
[24:16:56] [01] [00:00:00] Building lang/quilc | quilc-1.26.0_1
[24:17:21] [01] [00:00:25] Finished lang/quilc | quilc-1.26.0_1: Failed: build
```

**Comment by bdrewery** [comment]
It would help if I had let it actually get to the build. It had crashed early due to an unrelated reason.

Yes I see the output too!

**Comment by bdrewery** [comment]
It appears to be something with `sbcl` finding the TTY.
https://github.com/sbcl/sbcl/search?q=tty might have the answer.
`lsof` shows it opens the tty on fd 3. Poudriere can leak this down to the children but in my test I ensured it did not leak.
```
sbcl    9458 portbuild    3u  VCHR                  1,36   0t4221     292 (like character special /dev/pts/23)
```

**Comment by bdrewery** [comment]
Using `setsid` when running `make build` in `port_build` fixes the problem. A mockup of this is below but requires adding `BUILD_DEPENDS+= setsid>0:sysutils/setsid` in ports somewhere.
```
diff --git src/share/poudriere/common.sh src/share/poudriere/common.sh
index dc8d238a9..f6b7220cf 100755
--- src/share/poudriere/common.sh
+++ src/share/poudriere/common.sh
@@ -3120,6 +3110,19 @@ _real_build_port() {
        fi

        for phase in ${targets}; do
+               local setsid
+               case "${PKGNAME}" in
+               *ccache*|pkg-*|setsid-*) setsid= ;;
+               *) setsid="/usr/local/bin/setsid" ;;
+               esac
+               case "${phase}" in
+               build|test) ;;
+               *) setsid= ;;
+               esac
                max_execution_time=${MAX_EXECUTION_TIME}
                phaseenv=
                JUSER=${jailuser}
@@ -3240,7 +3243,7 @@ _real_build_port() {
                        nohang ${max_execution_time} ${NOHANG_TIME} \
                                ${log}/logs/${PKGNAME}.log \
                                ${MASTERMNT}/.p/var/run/${MY_JOBID:-00}_nohang.pid \
-                               injail /usr/bin/env ${pkgenv} ${phaseenv} ${PORT_FLAGS} \
+                               injail ${setsid} /usr/bin/env ${pkgenv} ${phaseenv} ${PORT_FLAGS} \
                                /usr/bin/make -C ${portdir} ${MAKE_ARGS} \
                                ${phase} 3>&- 4>&-
                        hangstatus=$? # This is done as it may return 1 or 2 or 3
```
The real fix won't use this port. Just a demo there.

**Comment by bdrewery** [comment]
`jexecd` was supposed to handle this issue forever ago but it did not work well with gathering ports metadata. We might be able to use it just for the build parts. #364 needs to be looked at. Some changes I made for gathering ports metadata should be reverted which may fix #364.
